aspnet/security/accconfirm.rst

```diff
@@ -1,236 +1,209 @@
+In this section we'll run the web app and show the account confirmation and password recovery flow.
 
 - Run the application and register a new user
 
 .. image:: accconfirm/_static/loginaccconfirm1.png
 
-- Check your email for the account confirmation link.
+- Check your email for the account confirmation link. If you don't get the email notification:
 
-- Click the link to confirm your email.
-
-.. image:: accconfirm/_static/loginaccconfirm2.PNG
+	* Check the SendGrid web site to verify your sent mail messages. 
+	* Check your spam folder.
+	* Try another email alias on a different email provider (Microsoft, Yahoo, Gmail, etc.)
+	* In SSOX, navigate to **dbo.AspNetUsers** and delete the email entry and try again.
 
+- Click the link to confirm your email.
 - Login with your email and password.
-
 - Log Off.
 
+Test password reset
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
 - Click Login and select **Forgot your password?**
+- Enter your email which was used to register the account with.
+- An email with the link to reset your password will be sent. Check your email and click it to reset your password.  After your password has been successfully reset, you can login with your email and new password.
+         
+Require email confirmation before login
+----------------------------------------
 
-.. image:: accconfirm/_static/loginaccconfirm3.PNG
+With the current templates, once a user completes the registration form, they are logged in. You generally want to confirm their email before logging them in. In the section below, we will modify the code to require new users have a confirmed email before they are logged in (authenticated). Update the ``HttpPost Login`` action with the following highlighted changes.
 
-- Enter your email which was used to register the account with.
+.. literalinclude:: accconfirm/sample/WebApplication1/src/WebApplication1/Controllers/AccountController.cs
+   :language: c#
+   :lines: 51-96
+   :emphasize-lines: 12-21
+   :dedent: 8
+
+- A security best practice is to not use production secrets in test and development. If you publish the app to Azure, you can set the SendGrid secrets as application settings in the Azure Web App portal. The configuration system is setup to read keys from environment variables.
 
-.. image:: accconfirm/_static/loginaccconfirm4.PNG
+Combine social and local login accounts
+-------------------------------------------
 
-- An email with the link to reset your password will be sent. Check your email and click it to reset your password.
+To complete this section, you must first enable an external authentication provider. See :doc:`sociallogins`.
 
-.. image:: accconfirm/_static/loginaccconfirm5.PNG
+You can combine local and social accounts by clicking on your email link. In the following sequence "RickAndMSFT@gmail.com" is first created as a local login, but you can create the account as a social log in first, then add a local login.
```

Comment:
We're required to use fictitious names and email addresses from an LCA-approved list. Let me know if you need help with this.